package com.rec.demo.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.ModelMap;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.http.HttpStatus;


import com.rec.exceptions.MiExcepcion;
import com.rec.services.UserService;


@Controller
@RequestMapping("/user")
public class UserController {

    @Autowired(required = false) // This means to get the bean called userRepository   // Which is auto-generated by Spring, we will use it to handle the data
    private UserService userService;
     
    @GetMapping("/registrar")  //localhost:8080/user/registrar
    public String registrar() {
        return "registration.html";
    }

    @GetMapping("/error")
    public String error(){
        return "error404.html";
    }

    @PostMapping("/registro")
    @ResponseStatus(HttpStatus.CREATED)
    public String registro(@RequestParam String username,
                            @RequestParam String password, 
                            ModelMap modelo) throws MiExcepcion{
        try {
            userService.createUser(username,password);
            modelo.put("exito", "USTED fue registrad@ exitosamente");
        } 
        catch (MiExcepcion ex) {
            modelo.put("error", ex.getMessage());
            return "registration.html";
        }

        return "index.html";
    }
}
